### ISSUE TITLE
Incorrect compiler diagnostic for hidden virtual member

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#9925]
The C# compiler produces an error for this

```cs
abstract class A
{
    public abstract void M();
}
abstract class B: A
{
    // it is an error to hide an inherited abstract member by 
    // member whose access level is public, protected internal, or protected(?)
    public int M = 0; // error: M.B hides inherited member A.M()
}
```

But the error is not justified by anything (that I could find) in the C# language specification. We should either describe the required error in the C# specification or modify the compiler to stop producing it.

@AlekseyTs @MadsTorgersen

**Comment by NoneGiven** [comment]
Probably a small typo in "M.B" instead of "B.M"

**Comment by gafter** [comment]
@NoneGiven Yes, thanks. Fixed.

**Comment by BillWagner** [comment]
@jaredpar @MadsTorgersen @AlekseyTs 

Is the correct fix for the standard to change, or the compiler to change behavior?  If it's the standard, I'll move to the dotnet/csharpstandard repo for the committee to address.

**Comment by AlekseyTs** [comment]
I think the error makes sense because it prevents the user from getting in a situation when an abstract member cannot be implemented because it is hidden.

**Comment by BillWagner** [comment]
moving to dotnet/csharpstandard
